pkg/collector/runner/runner.go

```diff
@@ -205,12 +214,14 @@ func (r *Runner) StopCheck(id check.ID) error {
 	defer r.m.Unlock()
 
 	if c, isRunning := r.runningChecks[id]; isRunning {
-		log.Debugf("Stopping check %s", c)
+		log.Debugf("Stopping check %s %s", string(id), c)
 		go func() {
+			// Remember that the check was stopped so that even if it runs we can discard its stats
 			c.Stop()
 			close(done)
 		}()
 	} else {
+		log.Debugf("check %s is not running", string(id))
```

Comment:
nit: capitalize first letter, clarify message (maybe add `not stopping it` or something similar), and use `c.ID()`